pkg/api/validation/validation.go

```diff
@@ -183,6 +183,13 @@ func ValidateEndpointsName(name string, prefix bool) (bool, string) {
 	return NameIsDNSSubdomain(name, prefix)
 }
 
+// ValidateEventName can be used to check whether the given event name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+func ValidateEventName(name string, prefix bool) (bool, string) {
+	return NameIsDNSSubdomain(name, prefix)
```

Comment:
"Non-backwards compatible desirable validation strengthening" is the term
I've used on others.

On Nov 12, 2015, at 4:37 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/api/validation/validation.go
<https://github.com/kubernetes/kubernetes/pull/15992#discussion_r44718333>:

> @@ -183,6 +183,13 @@ func ValidateEndpointsName(name string, prefix bool) (bool, string) {
>  	return NameIsDNSSubdomain(name, prefix)
>  }
>
> +// ValidateEventName can be used to check whether the given event name is valid.
> +// Prefix indicates this name will be used as part of generation, in which case
> +// trailing dashes are allowed.
> +func ValidateEventName(name string, prefix bool) (bool, string) {
> +	return NameIsDNSSubdomain(name, prefix)

Oh, I think @smarterclayton <https://github.com/smarterclayton> is talking
about the DNSSubdomain part. I don't consider that a regression, but it
might be worth an announcement as it could potentially break currently
working clients.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/15992/files#r44718333>.